src/components/EventsList/EventsList.js

```diff
@@ -1,58 +1,126 @@
-import React from "react";
-import { SubTitle, Title } from "../../assets/styles/shared.styled";
+import React, { useContext, useState, useEffect } from "react";
 
-import EventCard from "../EventCard/EventCard";
-import Filter from "../Filter/Filter";
-import SearchForm from "../Search/SearchForm";
-import Sort from "../Sort/Sort";
+import { EventsContext } from "../../EventsContext";
+
+import axios from "axios";
+
+import { MdPeople, MdLocationOn, MdEventNote } from "react-icons/md";
+import { CgDetailsMore } from "react-icons/cg";
+import { BsThreeDots } from "react-icons/bs";
+
+import { SubTitle, Title } from "../../assets/styles/shared.styled";
 import {
-  AllEvents,
   AllEventsAndActions,
   SortFilter,
-  StyledActions,
   StyledDetails,
   StyledEventsList,
 } from "./EventsList.styled";
 
+import EventCard from "../EventCard/EventCard";
+import Filter from "../Filter/Filter";
+import SearchForm from "../Search/SearchForm";
+import Sort from "../Sort/Sort";
+
 const EventsList = () => {
+  const [events, setEvents] = useContext(EventsContext);
+  const [selectedEvent, setSelectedEvent] = useState(null);
+
+  useEffect(() => {
+    getEventsList();
+  }, []);
+
+  const updateLoadedEvents = (data) => {
+    const loadedEvents = [];
+    for (const key in data) {
+      loadedEvents.push({
+        id: key,
+        title: data[key].title,
+        attendees: data[key].attendees,
+        location: data[key].location,
+        description: data[key].description,
+        date: data[key].date,
+        time: data[key].time,
+      });
+    }
+    setEvents(loadedEvents);
+  };
+
+  const getEventsList = () => {
+    axios
+      .get(
+        "https://react-events-app-7e674-default-rtdb.europe-west1.firebasedatabase.app/events.json"
+      )
+      .then((response) => {
+        console.log(response.data);
+        updateLoadedEvents(response.data);
+      })
+      .catch((error) => {
+        // handle error
+        console.log(error);
+      });
+  };
+
+  const handleShowDetails = (id) => {
+    const selected = events.filter((eventItem) => eventItem.id === id);
```

Comment:
this will return an array with only one element (selectedEvent). I think you need only that element.

Read this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find